interceptor/intercept.go

```diff
@@ -93,13 +94,42 @@ func (i *Interceptor) Intercept(nextHop string, reqPaymentHash []byte, reqOutgoi
 
 		if channelPoint == nil {
 			if bytes.Equal(paymentHash, reqPaymentHash) {
+				// TODO: When opening_fee_params is enforced, turn this check in a temporary channel failure.
+				if params == nil {
+					log.Printf("DEPRECATED: Intercepted htlc with deprecated fee mechanism.")
+					params = &OpeningFeeParams{
+						MinMsat:              uint64(i.config.ChannelMinimumFeeMsat),
+						Proportional:         uint32(i.config.ChannelFeePermyriad / 100),
+						ValidUntil:           time.Now().UTC().Add(time.Duration(time.Hour * 24)).Format(basetypes.TIME_FORMAT),
+						MaxIdleTime:          uint32(i.config.MaxInactiveDuration / 600),
+						MaxClientToSelfDelay: uint32(10000),
+					}
+				}
+
 				if int64(reqIncomingExpiry)-int64(reqOutgoingExpiry) < int64(i.config.TimeLockDelta) {
 					return InterceptResult{
 						Action:      INTERCEPT_FAIL_HTLC_WITH_CODE,
 						FailureCode: FAILURE_TEMPORARY_CHANNEL_FAILURE,
 					}, nil
 				}
 
+				validUntil, err := time.Parse(basetypes.TIME_FORMAT, params.ValidUntil)
+				if err != nil {
+					log.Printf("time.Parse(%s, %s) failed. Failing channel open: %v", basetypes.TIME_FORMAT, params.ValidUntil, err)
+					return InterceptResult{
+						Action:      INTERCEPT_FAIL_HTLC_WITH_CODE,
+						FailureCode: FAILURE_TEMPORARY_CHANNEL_FAILURE,
+					}, nil
+				}
+
+				if time.Now().UTC().After(validUntil) {
```

Comment:
I think at this point we want to check the current fee and if it is bellow the fee in the promise we should allow to move forward with the promised fee.